### PR TITLE
[Blazor WebAssembly standalone] Don't cache `index.html` during development

### LINKED ISSUE #59276: [Blazor WebAssembly] `dotnet watch` may not inject `aspnetcore-browser-refresh.js` after `dotnet run`
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

`dotnet watch` may not inject the `aspnetcore-browser-refresh.js` script for Blazor WebAssesmbly standalone apps. This is because the cached result of `index.html` gets used, and this bypasses the script injection middleware.

For example, if you run a default Blazor WebAssembly standalone app first using `dotnet run`, then the response for `index.html` will not include the browser refresh script (which is expected). Subsequently running `dotnet watch` will result in that same `index.html` getting used (without the injected script). Clearing the browser cache results in the script injection middleware re-running, and hot reload will work as expected.

Note that:
* This bug does _not_ affect Blazor Web Apps
* This _does_ repro when running a .NET 8 app using either the .NET 9 or .NET 8 SDK.

### Expected Behavior

Caching does not interfere with hot reload script injection, and `dotnet watch` works regardless of whether `dotnet run` was executed prior to it.

### Steps To Reproduce

1. `dotnet new blazorwasm`
2. `dotnet run`
3. Open the app in the browser
4. Stop the app (`ctrl+c`)
5. `dotnet watch`
6. Open the app in the browser again
7. Observe that hot reload does not work
8. Observe that the `aspnetcore-browser-refresh.js` script _was **not**_ injected into the document
10. Clear the browser's cache (Dev Tools -> Application -> Clear site data)
11. Refresh the page
12. Observe that the `aspnetcore-browser-refresh.js` script _**was**_ injected into the document

### Exceptions (if any)

_No response_

### .NET Version

9.0.200-preview.0.24529.19

### Anything else?

Was also able to repro using .NET SDK version 8.0.404.

<details>

<summary>
Output of dotnet --info
</summary>

```text
.NET SDK:
 Version:           9.0.200-preview.0.24529.19
 Commit:            3980fbc52d
 Workload version:  9.0.200-manifests.c6f19616
 MSBuild version:   17.13.0-preview-24529-12+ee511a938

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.200-preview.0.24529.19\

.NET workloads installed:
 [tvos]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    18.0.9617/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.tvos\18.0.9617\WorkloadManifest.json
   Install Type:              Msi

 [macos]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    15.0.9617/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.macos\15.0.9617\WorkloadManifest.json
   Install Type:              Msi

 [maui-windows]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    9.0.0/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.maui\9.0.0\WorkloadManifest.json
   Install Type:              Msi

 [maccatalyst]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    18.0.9617/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.maccatalyst\18.0.9617\WorkloadManifest.json
   Install Type:              Msi

 [ios]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    18.0.9617/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.ios\18.0.9617\WorkloadManifest.json
   Install Type:              Msi

 [android]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    35.0.7/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.sdk.android\35.0.7\WorkloadManifest.json
   Install Type:              Msi

 [wasm-tools-net6]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    9.0.0/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.workload.mono.toolchain.net6\9.0.0\WorkloadManifest.json
   Install Type:              Msi

 [wasm-tools-net7]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    9.0.0/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.workload.mono.toolchain.net7\9.0.0\WorkloadManifest.json
   Install Type:              Msi

 [wasm-tools-net8]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    9.0.0/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.workload.mono.toolchain.net8\9.0.0\WorkloadManifest.json
   Install Type:              Msi

 [wasm-tools]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    9.0.0/9.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\9.0.100\microsoft.net.workload.mono.toolchain.current\9.0.0\WorkloadManifest.json
   Install Type:              Msi

 [aspire]
   Installation Source: VS 17.13.35517.34
   Manifest Version:    8.2.2/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.2.2\WorkloadManifest.json
   Install Type:              Msi

Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.0
  Architecture: x64
  Commit:       9d5a6a9aa4

.NET SDKs installed:
  8.0.404 [C:\Program Files\dotnet\sdk]
  9.0.100 [C:\Program Files\dotnet\sdk]
  9.0.200-preview.0.24529.19 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.35 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0-rc.2.24474.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.35 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0-rc.2.24473.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.35 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 9.0.0-rc.2.24474.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

</details>

### PR DESCRIPTION
**Pull request by MackinnonBuck**
# [Blazor WebAssembly standalone] Don't cache `index.html` during development

Fixes an issue where the WebAssembly dev server (used during Blazor WebAssembly standalone development) returned `index.html` with incorrect caching headers, resulting in hot reload script injection sometimes being bypassed.

## Description

The fix is to add the `Cache-Control: no-store` header to the response for `index.html` when running the app via the WebAssembly dev server. This prevents the browser from using a cached response, which bypasses the script injection middleware.

Fixes #59276

### CODE REVIEW

File: src/Assets/build/Microsoft.AspNetCore.App.Internal.Assets.targets [hunk 1]
```diff
@@ -40,7 +40,10 @@
       <_FrameworkStaticWebAssetCandidate Remove="@(_MissingFrameworkStaticWebAssetCandidate)" />
     </ItemGroup>
 
-    <Message Importance="High" Text="Framework asset '%(_MissingFrameworkStaticWebAssetCandidate.Identity)' could not be found and won't be included in the project." />
+    <Message
```
Comment (MackinnonBuck) [inline src/Assets/build/Microsoft.AspNetCore.App.Internal.Assets.targets:43]: This is unrelated to the main fix - it was logging a message even when `_MissingFrameworkStaticWebAssetCandidate` was empty.

**Review by javiercn — APPROVED** [review]
